Packages/MovieDBUI/Sources/MovieDBUI/UI/Navigation/Coordinator.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  Coordinator.swift
+//  MovieDB
+//
+//  Created by Mouad Bj on 15/3/2023.
+//
+
+import Foundation
+import UIKit
+
+public protocol Coordinator {
+  
+  var parentCoordinator: Coordinator? { get set }
```

Comment:
Yeah, the topic of navigation is debatable, I am personally not a big fan of the coordinator pattern, but I added it into this source code just to showcase how we can plug a coordinator into the architecture answering a question I had in the last meeting. However, I agree that the navigation controller shouldn't be part of the coordinator protocol. I am not here to defend the pattern, but my understanding of it is that the protocol should have a reference of its children coordinators ([reference](https://khanlou.com/2015/01/the-coordinator/)), even if a leaf wouldn't have children (now), but it could have children tomorrow.
I would like to know your approach on navigation management.

Reply:
Indeed, we can discuss other `Navigation` solutions, ones that can also be adapted easily with SwiftUI navigation in the future.